### PR TITLE
numberfield added with error labeling

### PR DESCRIPTION
**Pull request by EVCareeria**

### CODE REVIEW

File: components/fields/number/NumberField.tsx [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+        props?.changeState,
+        props?.value,
+        props?.model?.required ?? false,
+        props?.model?.minValue,
+        props?.model?.maxValue,
+        toNumber,
+        stringifyInteger
+    );
+
+    const onChangeHandler = (event:any) =>  {
```
Comment (thejhh) [inline components/fields/number/NumberField.tsx:62]: This should use `useCallback`. 

Also, move it as a new reusable custom hook instead under the [`hooks/fields/number`](https://github.com/heusalagroup/fi.hg.frontend/tree/main/hooks/field/number).

File: components/fields/number/NumberField.tsx [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+    const styleScheme = props?.style ?? ThemeService.getStyleScheme();
+    const placeholder = props.placeholder ?? props.model?.placeholder ?? DEFAULT_PLACEHOLDER;
+    const label = props.label ?? props.model?.label ?? '';
+    const [decimalValue, setDecimalValue] = useState<string>('');
+    const [isValid, setIsValid] = useState<boolean>(true);
+
+    const {
+        fieldState,
+        value = decimalValue,
+        onChangeCallback
```
Comment (thejhh) [inline components/fields/number/NumberField.tsx:48]: We probably should have `useIntegerField` instead, that gives `onChangeCallback()` that's optimized for it, and this normal would give the callback below.

File: components/fields/number/NumberField.tsx [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+            setIsValid(false);
+        }else if (regexValidation.test(eventVal) || eventVal.length <= 1){
+            setIsValid(true);
+        } else {
+            setIsValid(false);
+        }
+        
+        onChangeCallback(event) // Only changes value if integer
+
+        toNumber(decimalValue) // returns false if decimal
```
Comment (thejhh) [inline components/fields/number/NumberField.tsx:83]: Hmm, where does it return it? Not from this function...

File: components/fields/number/NumberField.tsx [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+    readonly children    ?: ReactNode;
+}
+
+ export function NumberField (props: NumberFieldProps) {
+
+    const className = props?.className;
+    const styleScheme = props?.style ?? ThemeService.getStyleScheme();
+    const placeholder = props.placeholder ?? props.model?.placeholder ?? DEFAULT_PLACEHOLDER;
+    const label = props.label ?? props.model?.label ?? '';
+    const [decimalValue, setDecimalValue] = useState<string>('');
```
Comment (thejhh) [inline components/fields/number/NumberField.tsx:42]: We should have `useIntegerField()` and `useNumberField()` and these states should be implemented inside `useNumberField()`.

However, I'm not yet sure if this implementation is very good... However, it anyway should be done inside a specialized hook (and that hook may or may not use different other hooks to implement it's logic.) It probably looks clearer how it should be done, once done so.

File: components/fields/number/NumberField.tsx [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+                onChange={onChangeHandler}
+                readOnly={ props?.change === undefined }
+            />
+            {props?.children}
+        </label>
+    );
+
+}
+
+function toNumber (value : string | undefined) : number | undefined {
```
Comment (thejhh) [inline components/fields/number/NumberField.tsx:122]: These functions should be inside a reusable hook and/or Utils class.

**Review by thejhh — CHANGES_REQUESTED** [review]
More refactoring required and possibly some design

**Comment by EVCareeria** [comment]
Managed to get the component working in-sync 'Had to call change(toNumber())) twice however', error style only turns text red however. Tested in proc. app

**Pull request closed without merging**